core/io/marshalls.cpp

```diff
@@ -1523,8 +1618,21 @@ Error encode_variant(const Variant &p_variant, uint8_t *r_buffer, int &r_len, bo
 
 						_encode_string(E.name, buf, r_len);
 
+						Variant value;
+
+						if (E.name == CoreStringNames::get_singleton()->_script) {
+							Ref<Script> script = obj->get_script();
+							if (script.is_valid()) {
+								String path = script->get_path();
+								ERR_FAIL_COND_V_MSG(path.is_empty() || !path.is_resource_file(), ERR_UNAVAILABLE, "Failed to encode a path to a custom script.");
```

Comment:
Well, this actually breaks encoding/decoding of builtin scripts, which which used to work.

Reply:
Fixed. This is because `is_resource_file()` does not allow sub-resources. But `ResourceLoader.load()` correctly loads built-in scripts.

https://github.com/godotengine/godot/blob/df78c0636d79c9545a283e0e2a926d623998cc27/core/string/ustring.cpp#L4623-L4625